### PR TITLE
feat(qwen): add sequence parallel support for Qwen3.5 linear attention

### PR DESCRIPTION
**Pull request by meichangsu1**
## Summary

This PR adds sequence parallel support for **Qwen3.5 linear attention** by ulysses.

### Main changes

- Import and validate flash-linear-attention kernels at runtime.
  - `causal_conv1d` from `fla.modules.convolution`
  - `chunk_gated_delta_rule` from `fla.ops.gated_delta_rule`
- Add SP helper utilities for Qwen3.5 linear attention:
  - sequence/head sharding transforms
  - local padding mask handling
  - local convolution weight slicing
  - packed `cu_seqlens` derivation from `real_position_ids` when needed
- Patch  Qwen3.5 linear-attention (gdn)forward .

## PR type
- [x] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

### Why
Qwen3.5 uses mixed layer types (full + linear attention). Existing SP support is robust for full attention, but lacked model-specific linear-attention SP execution and kernel integration. This PR fix it.

### Scope
- Qwen3.5 linear attention SP path (`sp_world_size > 1`)
- FLA kernel dependency and runtime validation
- Training/prefill path support

### Limitations (explicit)
- `rp_world_size > 1` is not supported for Qwen3.5 linear attention.
- Decode path with cached recurrent states is not supported in this SP patch.

## Experiment results

Paste your experiment result here(if needed).
Train:   1%|          | 1/189 [02:41<8:25:06, 161.21s/it]

{'loss': '1.461', 'grad_norm': '1.705', 'learning_rate': '1e-05', 'token_acc': '0.6399', 'epoch': '0.016', 'global_step/max_steps': '1/189', 'elapsed_time': '2m 41s', 'remaining_time': '8h 25m 20s', 'memory(GiB)': '67.95', 'train_speed(s/it)': '161.3'}
{'loss': '1.484', 'grad_norm': '1.666', 'learning_rate': '2e-05', 'token_acc': '0.6324', 'epoch': '0.032', 'global_step/max_steps': '2/189', 'elapsed_time': '3m 55s', 'remaining_time': '6h 6m 35s', 'memory(GiB)': '68.05', 'train_speed(s/it)': '117.6'}
{'loss': '1.477', 'grad_norm': '1.767', 'learning_rate': '3e-05', 'token_acc': '0.6311', 'epoch': '0.048', 'global_step/max_steps': '3/189', 'elapsed_time': '4m 48s', 'remaining_time': '4h 57m 21s', 'memory(GiB)': '68.05', 'train_speed(s/it)': '95.92'}
{'loss': '1.498', 'grad_norm': '1.683', 'learning_rate': '4e-05', 'token_acc': '0.6355', 'epoch': '0.064', 'global_step/max_steps': '4/189', 'elapsed_time': '5m 36s', 'remaining_time': '4h 18m 56s', 'memory(GiB)': '68.05', 'train_speed(s/it)': '83.98'}
{'loss': '1.422', 'grad_norm': '1.528', 'learning_rate': '5e-05', 'token_acc': '0.6329', 'epoch': '0.08', 'global_step/max_steps': '5/189', 'elapsed_time': '6m 17s', 'remaining_time': '3h 51m 17s', 'memory(GiB)': '68.05', 'train_speed(s/it)': '75.42'}
{'loss': '1.322', 'grad_norm': '1.203', 'learning_rate': '6e-05', 'token_acc': '0.6457', 'epoch': '0.096', 'global_step/max_steps': '6/189', 'elapsed_time': '6m 53s', 'remaining_time': '3h 30m 1s', 'memory(GiB)': '68.05', 'train_speed(s/it)': '68.86'}
...
Train:  24%|██▍       | 46/189 [17:53<45:13, 18.97s/it]
{'loss': '1.049', 'grad_norm': '0.6095', 'learning_rate': '9.035e-05', 'token_acc': '0.7018', 'epoch': '0.736', 'global_step/max_steps': '46/189', 'elapsed_time': '17m 54s', 'remaining_time': '55m 38s', 'memory(GiB)': '68.06', 'train_speed(s/it)': '23.34'}

### CODE REVIEW

File: swift/model/models/qwen.py [hunk 1]
```diff
@@ -1131,11 +1155,272 @@ def get_model(self, model_dir: str, config, processor, model_kwargs) -> PreTrain
...
+
+    if sp_enabled:
+        core_attn_out = head_to_seq_shard(core_attn_out, sequence_parallel_context)
+    core_attn_out = mod.norm(core_attn_out.reshape(-1, mod.head_v_dim), z.reshape(-1, mod.head_v_dim))
+    core_attn_out = core_attn_out.reshape(batch_size, seq_len, local_value_dim if not sp_enabled else mod.value_dim)
+    return mod.out_proj(core_attn_out)
+
+
+def _patch_qwen3_5_linear_attention_sequence_parallel() -> None:
+    if os.environ.get('QWEN35_SP_LINEAR_HEAD_PARALLEL', '1') != '1':
```
Comment (Jintao-Huang) [inline swift/model/models/qwen.py]: 这个参数默认关闭吧，然后在args中 手动打开

Comment (meichangsu1) [reply swift/model/models/qwen.py]: 我直接去掉了，patch与否用是否启用sp 判断（--sequence_parallel_size  xxx）,这个环境变量多余了

**Review by Jintao-Huang — APPROVED** [review]

**Comment by Jintao-Huang** [comment]
thanks!
